packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -190,11 +201,216 @@ const renderQueryEditorHeaderLabel = (
+                              className="table__cell--left"
+                            >
+                              {column}
+                            </th>
+                          ))}
+                        </tr>
+                      </thead>
+                      <tbody>
+                        {(isMineOnly
+                          ? editorStore.queries.filter(
+                              (q) => q.isCurrentUserQuery,
+                            )
+                          : editorStore.queries
+                        ).map((query) => (
+                          <tr
+                            key={query.id}
+                            className={clsx(
+                              'query-editor__query-loader__body__table__row',
+                              {
+                                'query-editor__query-loader__body__table__row--selected':
+                                  selectedQueryID === query.id,
+                              },
+                            )}
+                            onClick={(event) => setSelectedQueryID(query.id)}
+                          >
+                            <td className="table__cell--left">{query.name}</td>
+                            <td className="table__cell--left">
+                              {query.owner ?? 'anonymous'}
+                            </td>
+                            <td className="table__cell--left">
+                              {query.versionId}
+                            </td>
+                          </tr>
+                        ))}
+                      </tbody>
+                    </table>
+                  </>
+                )}
+                {editorStore.queries.length === 0 && (
+                  <BlankPanelContent>No query available</BlankPanelContent>
+                )}
+              </>
+            )}
+            {!editorStore.loadQueriesState.hasCompleted && (
+              <>
+                <PanelLoadingIndicator
+                  isLoading={!editorStore.loadQueriesState.hasCompleted}
+                />
+                <BlankPanelContent>Loading queries...</BlankPanelContent>
+              </>
+            )}
+          </div>
+          <div className="search-modal__actions">
+            <button className="btn btn--dark" onClick={loadSelectedQuery}>
+              Load Query
+            </button>
+            <button className="btn btn--dark" onClick={closeQueryImporter}>
+              Close
+            </button>
+          </div>
+        </div>
+      </Dialog>
+    );
+  },
+);
+
 const QueryEditorHeaderContent = observer(
   (props: { queryBuilderState: QueryBuilderState }) => {
     const { queryBuilderState } = props;
     const editorStore = useQueryEditorStore();
     const applicationStore = useLegendQueryApplicationStore();
+    const openQueryLoader = (): void => {
+      editorStore.setIsQueryLoaderOpen(true);
+      flowResult(editorStore.loadQueries(''))
```

Comment:
also, it's fine to make this call here for the first time we bring this loader up, but see my note on mimicking what we do in `EditExistingQuerySetup`, we will do this in the `useEffect` there.